askama_parser/src/node.rs

```diff
@@ -563,25 +563,44 @@ impl<'a> Macro<'a> {
 #[derive(Debug, PartialEq)]
 pub struct FilterBlock<'a> {
     pub ws1: Ws,
-    pub filter_name: &'a str,
-    pub args: Vec<Expr<'a>>,
+    pub filters: Expr<'a>,
     pub nodes: Vec<Node<'a>>,
     pub ws2: Ws,
 }
 
 impl<'a> FilterBlock<'a> {
     fn parse(i: &'a str, s: &State<'_>) -> ParseResult<'a, Self> {
+        #[allow(clippy::type_complexity)]
+        fn filter(i: &str, level: crate::Level) -> ParseResult<'_, (&str, Option<Vec<Expr<'_>>>)> {
+            let (i, (_, fname, args)) = tuple((
+                char('|'),
+                ws(identifier),
+                opt(|i| Expr::arguments(i, level, false)),
+            ))(i)?;
+            Ok((i, (fname, args)))
+        }
```

Comment:
Can we reuse this across `Expression::filtered` and `FilterBlock`?

Reply:
I was wondering about that if I should wait for your other PR to be merged first. Merging both functions.